sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.CSharp8.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public interface IWithDefaultImplementation
+    {
+        void DoSomething();
+
+        //Default interface methods
+        void DoNothing()
+        {
+            var a = false;
+            if (a)  // Noncompliant
+            {       // Secondary
+                DoSomething(); // never executed
+            }
+        }
+    }
+
+    public class CSharp8
+    {
+        int SwitchExpression()
+        {
+            var a = false;
+            return a switch { true => 0, false => 1 };  // FN - switch arms are not inspected
```

Comment:
side notes:
- the switch arms are inspected but the only constrain we currently add is for null
- we have the same behavior also for switch statements so if we want to support this it would make sense to handle them together